Meta/CMake/serenity_toolchain.cmake

```diff
@@ -0,0 +1,30 @@
+#
+# Define and generate CMake toolchain to compile serenity with
+#
+
+set(SERENITY_ARCH "i686" CACHE STRING "Target architecture for SerenityOS (i686 or x86_64)")
+if("${SERENITY_ARCH}" STREQUAL "i686")
+    set(SERENITY_CLANG_ARCH "i386")
+else()
+    set(SERENITY_CLANG_ARCH ${SERENITY_ARCH})
+endif()
+
+if (${CMAKE_HOST_SYSTEM_NAME} MATCHES SerenityOS)
+    message("Good job on building cmake!")
+else()
+    # Note: We must generate the toolchain before calling project, as project looks for the toolchain entries
+    set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compiler toolchain to use for Serenity (GNU or Clang)")
+    set(SERENITY_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+    set(SERENITY_SOURCE_DIR "${CMAKE_SOURCE_DIR}")
+
+    configure_file("Toolchain/CMake/${SERENITY_TOOLCHAIN}Toolchain.txt.in" CMakeToolchain.txt @ONLY)
+    set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_BINARY_DIR}/CMakeToolchain.txt" CACHE PATH "Toolchain file to use for cross-compilation")
+
+    # Note: This is the price we pay for not requiring the toolchain to be generated externally
+    #       If the toolchain is specified via command line, we get its local variables imported into our CMake variables
```

Comment:
This note is kind of silly now with the superbuild structure. You have plenty of opportunities to generate the toolchain and pass is to the build that needs it.